webapp/src/components/Analysis/PerturbationTestingPreview.test.tsx

```diff
@@ -0,0 +1,167 @@
+      expect(screen.getByText(/16.0%/i)).toBeVisible();
+      expect(screen.getByText("Failure rate - Training Set")).toBeVisible();
+
+      // styling should be applied for the selected toggle as expected
+      expect(screen.getByText(/14.0%/)).toHaveStyle(
+        "font-size: 3.75rem; color: rgb(237, 108, 2)"
+      );
+
+      // verify with some random unexpected values to not have been displayed.
+      expect(screen.queryByText(/17.0%/i)).toBeNull();
+      expect(screen.queryByText("some random text")).toBeNull();
+      expect(screen.queryByText(/26.0%/i)).toBeNull();
+    });
+  });
+
+  it("should not display a value if datasplit is not available", async () => {
+    renderPerturbationTestingPreview({ train: false, eval: true });
+    await waitFor(() => expect(screen.getByText("--%")).toBeVisible());
+  });
+
+  it("should have expected columns", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    const expectedColumnHeaders = [
+      "Test Family",
+      "Test Name",
+      "Modif. Type",
+      "Test Description",
+      "FR on Evaluation Set",
+    ];
+    const actualColumnHeaders = screen.getAllByRole("columnheader");
+    await waitFor(() => {
+      expectedColumnHeaders.forEach((name, index) => {
+        expect(actualColumnHeaders[index].textContent).toEqual(name);
+      });
+    });
+  });
+
+  it("should have default sort as desc for colum header 'FR on Evaluation/Training set'", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    await waitFor(() =>
+      expect(
+        screen.getAllByRole("columnheader")[4].getAttribute("aria-sort")
+      ).toBe("descending")
+    );
+  });
+
+  it("should modify the last Column 'Failure Rate' and tooltips of it if the toggle is changed to train/eval", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
+    // before toggle changes
+    expect(screen.getAllByRole("columnheader")[4].textContent).toBe(
+      "FR on Evaluation Set"
+    );
+    expect(
+      screen.getByLabelText("Failure Rate on Evaluation Set")
+    ).toBeInTheDocument();
+    // after toggle changes
+    fireEvent.click(screen.getByRole("button", { pressed: false }));
+    await waitFor(() => {
+      expect(screen.getAllByRole("columnheader")[4].textContent).toBe(
+        "FR on Training Set"
+      );
+      expect(
+        screen.getByLabelText("Failure Rate on Training Set")
+      ).toBeInTheDocument();
+    });
+  });
+});
+
+describe("PerturbationTestingPreview without failure rate for training set", () => {
+  const handlers = [getPerturbationResponseWithoutTrainFailureRate];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+  it("should not display value if failure rate for a datasplit is not available", async () => {
+    renderPerturbationTestingPreview({ train: true, eval: true });
```

Comment:
I think we would not get a failure rate for training set only if the training set is not available. You will see that in that case, the `"--%"` should not have success color. The value not being available is neither good nor bad.
```suggestion
    renderPerturbationTestingPreview({ train: false, eval: true });
```

Reply:
so the purpose of this section was to have "--%" if dataset split is unavailable. My description is not appropriate here. is that what you are explaining here? and yes, we wont be having success color here.

so changing the description to ->  should display '--%' if a dataset split is unavailable. will this resolve the conversation?

Reply:
No, if the training dataset split is unavailable, it should be changed to `train: false` as I suggested.